src/main/java/Duke.java

```diff
@@ -1,3 +1,54 @@
+import java.util.Scanner;
+public class Duke {
+    public static void printLine() {
+        System.out.println("____________________________________________________________");
+    }
+    public static void main(String[] args) {
+        String[] list = new String[100];
+        String[] status = new String[100];
+        //initializing the array
+        for (int i = 0; i < 100; i++) {
+            list[i] = " ";
+            status[i] = " ";
+        }
+        int index = 0;
+        Scanner in = new Scanner(System.in);
+        printLine();
+        System.out.println("Hello! I'm Doraemon!");
+        System.out.println("What can I do for you?");
+        printLine();
+        while (!list[index].equals("bye")) {
+            list[index] = in.nextLine();
+            if (list[index].equals("bye")) {
+                break;
+            }
+            if (list[index].equals("list")) {
+                int counter = 1; //the numbering in the list
+                for (int j = 0; j < index; j++) {
+                    System.out.print(counter + ".[" + status[j] + "] ");
+                    System.out.println(list[j]);
+                    counter++;
+                }
+                index--;
+            } else if (list[index].contains("mark")) {
+                list[index] = list[index].substring(5);
```

Comment:
Could try to define the 5 so that it does not seem to be a "magic number" same thing applies to other areas such as 100 being the max String array size